test/test_loader.py

```diff
@@ -0,0 +1,91 @@
+import chess
+import numpy as np
+import torch.multiprocessing as mp
+from collections import deque
+import pickle
+import unittest
+import torch
+
+import os,sys,inspect
+parentdir = os.path.dirname(os.path.realpath(""))
+sys.path.insert(0,parentdir)
+
+from loader import Preprocessor, FENGenerator, PGNIterator, StandardConvSuite 
+
+path_to_test_pgn = f"lichessTestPgn.pgn"
+path_to_test_fen = f"lichessTestFen.pgn"
+path_to_test_array = f"lichessTestArray.dictionary"
+path_to_test_tensor = f"lichessTestTensor.dictionary"
+
+
+class TestIterator:
+    def __init__(self):
+        self.generator = FENGenerator(mp.Queue(maxsize=4), path_to_test_pgn)
+
+    def __iter__(self):
+        yield [self.generator.get_position()]
+
+    def close(self):
+        self.generator.closeFile()
+
+
+class TestSum(unittest.TestCase):
+    def test_FENGenerator(self):
+        self.generator = FENGenerator(mp.Queue(maxsize=4), path_to_test_pgn)
+        self.answers = open(path_to_test_fen, "r")
+        self.index = 0
+        for answer in self.answers:
+            self.index += 1
+            self.data_from_generator = self.generator.get_position()
+            self.assertEqual(
+                self.data_from_generator,
+                answer.replace("\n", ""),
+                "mistake in FENGenerator class, in example " + str(self.index),
+            )
+        self.answers.close()
+        self.generator.closeFile()
+        print("Endend")

```

Comment:
if there's a reason to leave this print in the code (is there?), make it understandable to whoever reads it